strawberry/enum.py

```diff
@@ -48,7 +64,16 @@ def _process_enum(
 
     description = description
 
-    values = [EnumValue(item.name, item.value) for item in cls]  # type: ignore
+    values = []
+    for item in cls:  # type: ignore
+        item_value = item.value
+        item_name = item.name
+        if isinstance(item_value, EnumValueDefinition):
+            values.append(
+                EnumValue(item_name, item_value.value, item_value.deprecation_reason)
+            )
+        else:
+            values.append(EnumValue(item_name, item_value, deprecation_reason=None))
```

Comment:
this might look slightly cleaner 😊

```suggestion
        item_value = item.value
        item_name = item.name
        deprecation_reason = None
        
        if isinstance(item_value, EnumValueDefinition):
            item_value = item_value.value
            deprecation_reason = item_value.deprecation_reason
        
        values.append(EnumValue(item_name, item_value, deprecation_reason=deprecation_reason))
```